lib/cfa/grub2/default.rb

```diff
@@ -102,19 +102,27 @@ def cryptodisk
           true_value: "y", false_value: "n")
       end
 
+      VALID_TERMINAL_OPTIONS = [:serial, :console, :gfxterm].freeze
+      # Reads value of GRUB_TERMINAL from /etc/default/grub
+      #
+      # GRUB_TERMINAL option allows multiple values as space separated string
+      #
+      # @return [Array<Symbol>, nil] an array of symbols where each symbol
+      #                              represents supported terminal definition
       def terminal
-        value = value_for("GRUB_TERMINAL")
-        case value
-        when "", nil   then nil
-        when "console" then :console
-        when "serial"  then :serial
-        when "gfxterm" then :gfxterm
-        else
-          raise "unknown GRUB_TERMINAL option #{value.inspect}"
+        values = value_for("GRUB_TERMINAL")
+
+        return nil if values.nil? || values.empty?
+
+        values.split.map do |value|
+          if VALID_TERMINAL_OPTIONS.include?(value.to_sym)
+            value.to_sym
+          else
+            raise "unknown GRUB_TERMINAL option #{value.inspect}"
+          end
         end
       end
 
-      VALID_TERMINAL_OPTIONS = [:serial, :console, :gfxterm].freeze
```

Comment:
ah, now I see that terminal= is not similar to terminal getter. This looks quite wrong for me. it should accept array or nil and behave same as terminal getter.